### PR TITLE
Add a hide from frontpage button to posts

### PR DESCRIPTION
**Pull request by steveryb**
This creates a "Hide/Unhide post from frontpage" button    
in the context menu of posts on the "Frontpage posts" and "All Posts"    
(to allow users to unhide) which will, as you'd expect, hide a post.    
    
This is a request from super users: sometimes they visit the forum quite    
often, and find their frontpage has the same content, making it hard    
to discover what's new. This allows users to filter their frontpage a    
bit, making it easier for them to find what's new.    
    
This button is only shown on the "Frontpage posts" and "All Posts"    
sections, and only hides posts in the "Frontpage posts" section. This    
hopefully makes it more clear that this won't ever e.g. let users hide    
pinned posts, and lets user recover an accidentally hidden post by going    
to all posts.    
    
There's three major things I'd have included if I had the time, tl;drs    
here, longer notes in the code:    
1) Revisit the data model. The current data model is fine, but could    
be a problem if users hide a lot of posts. A model similar to    
`ReadStatuses` is likely preferable.    
2) Fix race condition in mutation. Unlikely to be hit in practice, but    
would be nice to always be reliable here.    
3) Move filtering to server side. If users hide a lot of posts, the    
frontpage becomes empty, or the pages become small.    

Screenshots:
| Description  | Screenshot |
| ------------- | ------------- |
| Option to hide from frontpage  | Screenshot from 2022-06-07 17-53-43|
| Option to unhide from all posts  | Screenshot from 2022-06-07 17-55-18 |
| No option to hide or unhide from pinned posts |Screenshot from 2022-06-07 17-55-54 |

### CODE REVIEW

File: packages/lesswrong/components/common/HomeLatestPosts.tsx [hunk 1]
```diff
@@ -37,6 +37,9 @@ const styles = (theme: ThemeType): JssStyles => ({
 
 const latestPostsName = forumTypeSetting.get() === 'EAForum' ? 'Frontpage Posts' : 'Latest'
 
+// Used to contain the context for hiding posts, so we don't need to thread through a variable through the hierarchy
+export const AllowHidingPostsContext = React.createContext<boolean>(false)
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/common/HomeLatestPosts.tsx:41]: My call is in favor of contexts.

**Comment by steveryb** [comment]
Sorry yeah! This PR will be up and running soon, but tl;dr: it adds a button to allow you to hide front page posts.

**Comment by steveryb** [comment]
This PR should be reviewable now! I might need to do some small changes to make tests happy, but ready to receive feedback. Updated the description so it should be more obvious what changed.

**Comment by jpaddison3** [comment]
Cypress tests have been flaky since the react 18 upgrade, sorry about that.

**Comment by jpaddison3** [comment]
Oh, maybe there's also a real error in there.

**Review by jpaddison3 — APPROVED** [review]

**Comment by Raemon** [comment]
This PR actually broke curated posts on lesswrong (I'm still looking into it, but looks like the changes to hideLastUnread didn't take into account the goals of the curated section. My guess is that hideLastUnread is actually makes more sense to think of as a separate project than this PR, one of it's purposes is to leave a single unread post remaining so that people can always see the latest curated post even if they read it already)

**Pull request merged**